packages/dd-trace/src/dogstatsd.js

```diff
@@ -31,14 +43,26 @@ class Client {
     this._add(stat, value, 'c', tags)
   }
 
-  flush () {
+  flush (cb) {
     const queue = this._enqueue()
 
     if (this._queue.length === 0) return
 
     this._queue = []
 
-    if (this._family !== 0) {
+    if (this._httpOptions) {
+      requestQueue(queue, this._httpOptions, err => {
+        if (err) {
+          log.debug('HTTP error from agent: ' + err.stack)
+          this._httpOptions = null
```

Comment:
This should actually only happen based on the status code. Socket errors mean that the agent couldn't be reached, in which case it might be down or completely unreachable. Only a 404 would confirm that the agent does not support HTTP for runtime metrics.

Reply:
What should be the behaviour when there's a socket error?